### PR TITLE
Implement structured logging with item-handler service

### PR DESCRIPTION
**Pull request by caismith-ch**
Update existing logging to implement structured logging.

Resolves: GCI-1079

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java [hunk 1]
```diff
@@ -101,8 +101,9 @@ public void processOrderReceivedError(org.springframework.messaging.Message<Orde
             handleMessage(message);
         }
         else {
-            LOGGER.info(String.format("Pausing error consumer \"%1$s\" as error recovery offset '%2$d' reached.",
-                    ORDER_RECEIVED_GROUP_ERROR, errorRecoveryOffset));
+            Map<String, Object> logMap = LoggingUtils.createLogMap();
+            logMap.put(ORDER_RECEIVED_GROUP_ERROR, errorRecoveryOffset);
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java:105]: CH Logging standards - the key name needs to be snake_case not kebab-case
e.g. 
error_recovery_offset
Also, if possible, can we log the topic name and offset of the message? errorRecovery Offset is a constant in this class, so will always be 0L

File: src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java [hunk 2]
```diff
@@ -238,8 +238,12 @@ protected Message createRetryMessage(String orderUri, String topic) {
         try {
             message.setValue(serializer.toBinary(orderReceived));
         } catch (SerializationException e) {
-            LOGGER.error(String.format("Error serializing message: \"%1$s\" for topic: \"%2$s\". \"%n%3$s\"",
-                    orderUri, topic, Arrays.toString(e.getStackTrace())));
+            Map<String, Object> logMap = LoggingUtils.createLogMap();
+            LoggingUtils.logIfNotNull(logMap, LoggingUtils.MESSAGE, orderUri);
+            LoggingUtils.logIfNotNull(logMap, LoggingUtils.TOPIC, topic);
+            logMap.put(LoggingUtils.EXCEPTION, e);
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java:244]: Can we log the message offset if possible

Comment (caismith-ch) [reply src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java]: Good shout!

**Comment by caismith-ch** [comment]
Comments addressed.

**Pull request merged**